### PR TITLE
Use transparent PNG for logo

### LINKED ISSUE #8158: [appearance][Dark Mode] Logo in README should have a transparent background
STR:
1. Enable Dark Mode on GitHub (https://github.com/settings/appearance)
2. Go to https://github.com/pytest-dev/pytest
3. See the pytest logo stand out with its white background while all the surroundings are dark:
pytest-readme-top

The root cause is that the image file does not have a transparent background: https://github.com/pytest-dev/pytest/blob/master/doc/en/img/pytest1.png. And I don't think anybody has its original source. So to fix this, somebody would need to do image editing or maybe image tracing — these days it sounds like a good idea to have an SVG instead.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #8158

### CODE REVIEW

File: README.rst
Comment (webknjaz) [inline README.rst:4]: This seems to be working too so no need to edit SVG maybe.

Comment (nicoddemus) [reply README.rst:4]: Yeah I didn't edit it. 👍

Comment (nicoddemus) [reply README.rst:4]: Bummer `align: center` doesn't seem to work though... 😕

Comment (webknjaz) [reply README.rst:4]: Yep, GH support for RST is quite limited.

File: README.rst
Comment (nicoddemus) [inline README.rst:1]: This needs to be updated before merging.

File: README.rst
Comment (nicoddemus) [reply README.rst:4]: If you are happy with the results, can you approve and then we can get this online? 😁

File: README.rst
Comment (webknjaz) [reply README.rst:1]: Also: you may want to update the `conf.py` to use SVG and see how it goes.

**Review by webknjaz — APPROVED** [review]
LGTM. Feel free to merge unless you want to follow-up on the other comments (like using SVG in Sphinx too or editing SVG to abuse its way of rendering to emulate centering)

**Comment by nicoddemus** [comment]
Yeah using the `svg` file on the docs looks good too:

image

**Comment by webknjaz** [comment]
@nicoddemus you'll need someone else's approval because I don't have the commit bit here, hence I can't unblock it for you.

cc @The-Compiler?

**Comment by nicoddemus** [comment]
> @nicoddemus you'll need someone else's approval because I don't have the commit bit here, hence I can't unblock it for you.

No worries, I can merge it as is. 👍

**Pull request merged**
